### DISCUSSION TITLE
Graphics not defined

### DISCUSSION THREAD

**Discussion by Softwareincorp** [ProjectPhysX/FluidX3D#135] [General]
Good afternoon.

To work with the block I needed, I uncommented the "river" area, but when I try to compile the program and run it, it gives the error `class "LBM" has no member "graphics".`
![image](https://github.com/ProjectPhysX/FluidX3D/assets/105108977/db655f43-7824-416b-b67f-484ce62c0bb4)
In the file `lbm.hpp` there is an import line `#include graphics.hpp"`. 
I tried to uncomment in this file lines:
`//#define INTERACTIVE_GRAPHICS`
`//#define INTERACTIVE_GRAPHICS_ASCII`
`//#define GRAPHICS`

And in this case the message changed to 

`_alloca indicates failure by raising a stack overflow exception. Consider using _malloca instead. FluidX3D\src\OpenCL\include\CL\cl.hpp 1146.`

 What should I do to compile the program without errors?

I uncommented all other necessary modules in the file "defines.hpp"

 
Thank you in advance.

![image](https://github.com/ProjectPhysX/FluidX3D/assets/105108977/db33ac5f-8191-4db6-baff-47fdbeef142a)

**Comment by ProjectPhysX** [comment]
Hi @Softwareincorp,

don't forget to comment out `//#define BENCHMARK`! The `BENCHMARK` define disables all other extensions, including `INTERACTIVE_GRAPHICS`.

From the graphics options, you only need to activate
```c
#define INTERACTIVE_GRAPHICS
```
as that in itself activates the Windows/Linux backends and `GRAPHICS`, which turns on the OpenCL rendering engine.

Kind regards,
Moritz

**Reply by Softwareincorp** [reply]
Thank you so much. It works.
